ucp/__init__.py

```diff
@@ -64,9 +64,13 @@ def _is_mig_device(handle):
             if _is_mig_device(handle):
                 continue
 
-            total_memory = pynvml.nvmlDeviceGetMemoryInfo(handle).total
-            bar1_total = pynvml.nvmlDeviceGetBAR1MemoryInfo(handle).bar1Total
+            try:
+                bar1_total = pynvml.nvmlDeviceGetBAR1MemoryInfo(handle).bar1Total
+            except pynvml.nvml.NVMLError_NotSupported:
+                # Bar1 access not supported on this device. Skip
+                continue
```

Comment:
@pentschev - Is it "correct" to leave the `large_bar1` element as `False` for devices without Bar1 access?